pennylane/drawer/_add_obj.py

```diff
@@ -0,0 +1,245 @@
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the _add_obj function for adding an object to the text drawer.
+"""
+# pylint: disable=unused-argument
+from functools import singledispatch
+
+import pennylane as qml
+from pennylane.measurements import (
+    CountsMP,
+    DensityMatrixMP,
+    ExpectationMP,
+    MidMeasureMP,
+    ProbabilityMP,
+    SampleMP,
+    StateMP,
+    VarianceMP,
+)
+
+
+def _add_cond_grouping_symbols(op, layer_str, config):
+    """Adds symbols indicating the extent of a given object for conditional
+    operators"""
+    n_wires = len(config.wire_map)
+
+    mapped_wires = [config.wire_map[w] for w in op.wires]
+    mapped_bits = [config.bit_map[m] for m in op.meas_val.measurements]
+    max_w = max(mapped_wires)
+    max_b = max(mapped_bits) + n_wires
+
+    ctrl_symbol = "╩" if config.cur_layer != config.cwire_layers[max(mapped_bits)][-1] else "╝"
+    layer_str[max_b] = f"═{ctrl_symbol}"
+
+    for w in range(max_w + 1, max(config.wire_map.values()) + 1):
+        layer_str[w] = "─║"
+
+    for b in range(n_wires, max_b):
+        if b - n_wires in mapped_bits:
+            intersection = "╣" if config.cur_layer == config.cwire_layers[b - n_wires][-1] else "╬"
+            layer_str[b] = f"═{intersection}"
+        else:
+            filler = " " if layer_str[b][-1] == " " else "═"
+            layer_str[b] = f"{filler}║"
+
+    return layer_str
+
+
+def _add_grouping_symbols(op, layer_str, config):  # pylint: disable=unused-argument
+    """Adds symbols indicating the extent of a given object."""
+
+    if len(op.wires) > 1:
+        mapped_wires = [config.wire_map[w] for w in op.wires]
+        min_w, max_w = min(mapped_wires), max(mapped_wires)
+        layer_str[min_w] = "╭"
+        layer_str[max_w] = "╰"
+
+        for w in range(min_w + 1, max_w):
+            layer_str[w] = "├" if w in mapped_wires else "│"
+
+    return layer_str
+
+
+def _add_mid_measure_grouping_symbols(op, layer_str, config):
+    """Adds symbols indicating the extent of a given object for mid-measure
+    operators"""
+    print("here")
+    if op not in config.bit_map:
+        return layer_str
+
+    n_wires = len(config.wire_map)
+    mapped_wire = config.wire_map[op.wires[0]]
+    bit = config.bit_map[op] + n_wires
+    layer_str[bit] += " ╚"
+
+    for w in range(mapped_wire + 1, n_wires):
+        layer_str[w] += "─║"
+
+    for b in range(n_wires, bit):
+        filler = " " if layer_str[b][-1] == " " else "═"
+        layer_str[b] += f"{filler}║"
+
+    return layer_str
+
+
+@singledispatch
+def _add_obj(
+    obj, layer_str: list[str], config, tape_cache=None, skip_grouping_symbols=False
+) -> list[str]:
```

Comment:
```suggestion
) -> list[str]: # pragma: no cover
```
I guess we don't need to cover this numb main default patch

Reply:
I'm going to add a quick test, since it should be easy enough to do.